### ISSUE TITLE
Error thrown when a external ledger is used

### ISSUE THREAD

**Issue by sgroh** [bcgov/vc-authn-oidc#381]
When an external ledger is used on Issuer-Kit and on VC-Authn-OIDC project and the usage of a pre-configured mobile wallet like Esatus (with the external ledger previously added)  the mobile application receives an error right after scan the QR code to login on the "vc-authn-oidc demo application". 
Please find the stacktrace error below:
`controller_1     | {"event": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py\", line 78, in call_next\n    message = await recv_stream.receive()\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/anyio/streams/memory.py\", line 97, in receive\n    return self.receive_nowait()\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/anyio/streams/memory.py\", line 90, in receive_nowait\n    raise EndOfStream\nanyio.EndOfStream\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/app/api/main.py\", line 90, in logging_middleware\n    response: Response = await call_next(request)\n                         ^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py\", line 84, in call_next\n    raise app_exc\n  File \"/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py\", line 70, in coro\n    await self.app(scope, receive_or_disconnect, send_no_error)\n  File \"/usr/local/lib/python3.11/site-packages/starlette/middleware/cors.py\", line 83, in __call__\n    await self.app(scope, receive, send)\n  File \"/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py\", line 79, in __call__\n    raise exc\n  File \"/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py\", line 68, in __call__\n    await self.app(scope, receive, sender)\n  File \"/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py\", line 21, in __call__\n    raise e\n  File \"/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py\", line 18, in __call__\n    await self.app(scope, receive, send)\n  File \"/usr/local/lib/python3.11/site-packages/starlette/routing.py\", line 718, in __call__\n    await route.handle(scope, receive, send)\n  File \"/usr/local/lib/python3.11/site-packages/starlette/routing.py\", line 276, in handle\n    await self.app(scope, receive, send)\n  File \"/usr/local/lib/python3.11/site-packages/starlette/routing.py\", line 66, in app\n    response = await func(request)\n               ^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/fastapi/routing.py\", line 237, in app\n    raw_response = await run_endpoint_function(\n                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/local/lib/python3.11/site-packages/fastapi/routing.py\", line 163, in run_endpoint_function\n    return await dependant.call(**values)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/app/api/routers/presentation_request.py\", line 50, in send_connectionless_proof_req\n    if 'text/html' in req.headers.get('accept'):\n       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nTypeError: argument of type 'NoneType' is not iterable\n", "logger": "api.main", "level": "error", "timestamp": "2023-10-30T17:19:41.924874Z"}
`

**Comment by sgroh** [comment]
Additional information: Both projects shared same  Keycloak instance, and the ngrok tunnels were started according to receive calls from the mobile application. Below an example of my configuration:
```
authtoken: <Your_Personal_Ngrok_Token>
tunnels:
  issuer-agent:
    addr: 8021
    proto: http
    bind-tls: true
  vc-authn-controller:
    addr: 5000
    proto: http
    bind-tls: true
  vc-authn-aca-py:
    addr: 8030
    proto: http
    bind-tls: true
version: "1"

```

**Comment by esune** [comment]
@sgroh did you add the new ledger to the list of ledgers the agent will be connecting to? See https://github.com/bcgov/vc-authn-oidc/blob/0ac2f324a6686c243c37135f8396bb88c3e6ba8d/docker/agent/config/ledgers.yaml

**Comment by sgroh** [comment]
@esune yes I did, actually I remove all and add only one pointing to my private ledger:
```
- id: My-Own-Von-Network
  is_production: true
  is_write: true
  genesis_url: 'https://<MY_HOST>/genesis'
```

Regarding the wallet I've also tried with Lissi Wallet and the error is exactly the same. 
As you mentioned the error is exactly when in code it asks if the request's headers has on accept 'text/html'... Seems to be blowing up right there.

**Comment by esune** [comment]
My guess is that both eSatus and Lissi (both based on `aries-framework-dotnet`) are not sending compatible request headers.
@popkinj do you have any input on whether we should be updating the code to handle things differently, or the changes should potentially come upstream in `aries-framework-dotnet` when making the request?

**Comment by popkinj** [comment]
> My guess is that both eSatus and Lissi (both based on `aries-framework-dotnet`) are not sending compatible request headers. @popkinj do you have any input on whether we should be updating the code to handle things differently ~, or the changes should potentially come upstream in `aries-framework-dotnet` when making the request~ ?

Yes.... It's possible the headers are slightly different. If the header is requesting JSON we assume the qrcode is being scanned by a wallet. If requesting HTML then it is most likely a regular camera app. Can we capture the headers in all scenarios so I can have a look :eyes: ? There could be something non-standard... or possibly something I didn't account for. :thinking:

**Comment by sgroh** [comment]
Hi @popkinj below are the headers that I could capture from the request previous the error:
"host" : "<MY-NGROK>-free.app"
"user-agent" : "Dalvik/2.1.0 (Linux; U; Android 13; SM-G998B Build/TP1A.220624.014"
"accept-encoding" : "identity"
"x-forwarded-for" : <MY-IP>"
"x-forwarded-host" : "<MY-NGROK>-free.app"
"x-forwarded-proto" : "https"

Can I help with any other input here?

**Comment by sgroh** [comment]
@popkinj you are welcome.
No, that information is from Esatus Wallet.

On Lissi Wallet things become a bit weird, right after I scan the QR code to login in the vc-authn-oidc demo application, the mobile application receives a Customer Card, but I never was redirected to the demo portal as a logged user.

**Comment by popkinj** [comment]
> @popkinj you are welcome. No, that information is from Esatus Wallet.
> 
> On Lissi Wallet things become a bit weird, right after I scan the QR code to login in the vc-authn-oidc demo application, the mobile application receives a Customer Card, but I never was redirected to the demo portal as a logged user.

That is a little strange.
Thanks for the feedback. I think I know what's going on. Will keep you posted.

**Comment by popkinj** [comment]
K... I've been able to duplicate the issue in the Esatus Wallet.... and can see what's happening. There's definitely no `accept` header. Working on a solution.

**Comment by esune** [comment]
@popkinj if it looks like the effort to fix this is disproportionate to the benefit, considering `aries-framework-dotnet` is not maintained anymore, let's reassess.

@sgroh one thing you could do is opening an issue in https://github.com/hyperledger/aries-mobile-agent-react-native requesting the a feature to be able to add/remove ledgers like eSatus/Lissi allow you to do - that could be interesting and beneficial for others too.

**Comment by popkinj** [comment]
It was a small fix. I'm including it in [374](https://github.com/bcgov/vc-authn-oidc/issues/374)

**Comment by esune** [comment]
@sgroh look out for the next release of vc-authn OR just pull and build from main once #374 is merged. I would still add the recommendation in Bifold as it seems like a good idea.

**Comment by esune** [comment]
> It was a small fix. I'm including it in [374](https://github.com/bcgov/vc-authn-oidc/issues/374)

@popkinj since #374 is taking more time than expected, can you please make a separate PR with the quick fix for this, so that we can merge it to main?
